mantidproject/mantid
# Sanitise Invalid Nexus Log Characters

Description of work.
This PR fixes an issue where invalid characters were being passed to `iscntrl`. We now check that the character is within a valid range (0 - 255) for the c style locale we use by default then continue to call `iscntrl`.

Additionally I've swapped out the c style library `cctypes` with `locale` and switched to `std::iscntrl`.

**To test:**
- Build on MSVC in debug mode (for the assertion to work):
- Load 
[IMAT00003680](https://github.com/mantidproject/mantid/files/821190/IMAT00003680.zip) on Master, ensure it throws a assertion error (note if you run without debugging Mantid will appear to close gracefully)
- Load the same file on this branch - it should load successfully
- Go to log data. Check the value for `EPICS_PUTLOG` does not contain special characters

Fixes #19062 .

Does not need to be in the release notes. - Assertion is only enabled for debug mode.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
